### ISSUE TITLE
'ionic lib update' is not working.

### ISSUE THREAD

**Issue by tomkrus007** [driftyco/ionic-cli#646]
I want to update ionic from v1.1.0 to v1.1.1,but it's not working. the ionic version is still 1.1.0.

$ ionic lib update
Are you sure you want to replace /Users/tom/workspace/esm/www/lib/ionic with an updated version of Ionic?
(yes/no): yes
Latest version: 1.1.0  (released 2015-08-13)
Downloading: https://github.com/driftyco/ionic-bower/archive/v1.1.0.zip
[=============================]  100%  0.0s
Ionic version updated to: 1.1.0

**Comment by TheMattRay** [comment]
I saw that "nice to have" was tagged/added on this. My question is this: If doing ionic lib update is not the proper way to upgrade a project to the latest version, what is?

**Comment by brandyscarney** [comment]
@TheMattRay There are some issues with the v1.1.1 update that we need to resolve but yes, `ionic lib update` is the best way to upgrade. 

In the meantime you should be able to run `bower install ionic#1.1.1` to get this version.

**Comment by mhanke** [comment]
Are the problems with the v1.1.1 available yet? I would like to update to 1.1.1 because of a bugfix.

**Comment by RahulChamle** [comment]
Hi i updated ionic lib from

**Comment by RahulChamle** [comment]
Hi i updated ionic lib from v1.0.0-beta.11 to  v1.2.4-nightly-1917 and i got the error

 Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
 Watchers fired in the last 5 iterations: []
 http://errors.angularjs.org/1.4.3/$rootScope/infdig?p0=10&p1=%5B%5D
     at file:///android_asset/www/lib/ionic/js/ionic.bundle.js:13380:12
     at Scope.$digest (file:///android_asset/www/lib/ionic/js/ionic.bundle.js:29033:19)
     at Scope.$apply (file:///android_asset/www/lib/ionic/js/ionic.bundle.js:29263:24)
     at done (file:///android_asset/www/lib/ionic/js/ionic.bundle.js:23676:47)
     at completeRequest (file:///android_asset/www/lib/ionic/js/ionic.bundle.js:23848:7)
     at <error: Error: INVALID_STATE_ERR: DOM Exception 11>:25642
 Uncaught Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
 Watchers fired in the last 5 iterations: []
 http://errors.angularjs.org/1.4.3/$rootScope/infdig?p0=10&p1=%5B%5D:29266
 Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
 Watchers fired in the last 5 iterations: []
 http://errors.angularjs.org/1.4.3/$rootScope/infdig?p0=10&p1=%5B%5D
     at file:///android_asset/www/lib/ionic/js/ionic.bundle.js:13380:12
     at Scope.$digest (file:///android_asset/www/lib/ionic/js/ionic.bundle.js:29033:19)
     at Scope.$apply (file:///android_asset/www/lib/ionic/js/ionic.bundle.js:29263:24)
     at done (file:///android_asset/www/lib/ionic/js/ionic.bundle.js:23676:47)
     at completeRequest (file:///android_asset/www/lib/ionic/js/ionic.bundle.js:23848:7)
     at <error: Error: INVALID_STATE_ERR: DOM Exception 11>:25642
 Uncaught Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
 Watchers fired in the last 5 iterations: []
 http://errors.angularjs.org/1.4.3/$rootScope/infdig?p0=10&p1=%5B%5D:29266


What to do now...

**Comment by filoxo** [comment]
I found that running running `ionic lib update` internally calls `bower update` and will install everything in a `bower_components/` directory at the project root. I added a `.bowerrc` file that simply declares the installation directory to put the packages in. 

    {
      "directory" : "www/lib"
    }

This can't be set in `bower.json` because... ¯\\_(ツ)_/¯. After you have this file, running `ionic lib update` or `bower install [package]` puts the new package files in the right place for my Ionic project.
